src/Analyzers/CSharp/Analyzers/UseAutoProperty/CSharpUseAutoPropertyAnalyzer.cs

```diff
@@ -28,83 +31,36 @@ protected override bool SupportsPropertyInitializer(Compilation compilation)
+            => node as PropertyDeclarationSyntax;
 
-        private void AnalyzeMembers(
-            SemanticModelAnalysisContext context,
-            SyntaxList<MemberDeclarationSyntax> members,
-            List<AnalysisResult> analysisResults)
+        protected override void RegisterIneligibleFieldsAction(
+            ConcurrentSet<IFieldSymbol> ineligibleFields, SemanticModel semanticModel, SyntaxNode codeBlock, CancellationToken cancellationToken)
         {
-            foreach (var memberDeclaration in members)
+            foreach (var argument in codeBlock.DescendantNodesAndSelf().OfType<ArgumentSyntax>())
             {
-                AnalyzeMemberDeclaration(context, memberDeclaration, analysisResults);
+                // An argument will disqualify a field if that field is used in a ref/out position.  
+                // We can't change such field references to be property references in C#.
+                if (argument.RefKindKeyword.Kind() != SyntaxKind.None)
+                    AddIneligibleFields(semanticModel, argument.Expression, ineligibleFields, cancellationToken);
             }
-        }
 
-        private void AnalyzeMemberDeclaration(
-            SemanticModelAnalysisContext context,
-            MemberDeclarationSyntax member,
-            List<AnalysisResult> analysisResults)
-        {
-            if (member is BaseNamespaceDeclarationSyntax namespaceDeclaration)
-            {
-                AnalyzeMembers(context, namespaceDeclaration.Members, analysisResults);
-            }
-            else if (member is TypeDeclarationSyntax(
-                SyntaxKind.ClassDeclaration or
-                SyntaxKind.StructDeclaration or
-                SyntaxKind.RecordDeclaration or
-                SyntaxKind.RecordStructDeclaration) typeDeclaration)
-            {
-                // If we have a class or struct, recurse inwards.
-                AnalyzeMembers(context, typeDeclaration.Members, analysisResults);
-            }
-            else if (member is PropertyDeclarationSyntax propertyDeclaration)
-            {
-                AnalyzeProperty(context, propertyDeclaration, analysisResults);
-            }
-        }
+            foreach (var refExpression in codeBlock.DescendantNodesAndSelf().OfType<RefExpressionSyntax>())
+                AddIneligibleFields(semanticModel, refExpression.Expression, ineligibleFields, cancellationToken);
 
-        protected override void RegisterIneligibleFieldsAction(
-            List<AnalysisResult> analysisResults, HashSet<IFieldSymbol> ineligibleFields,
-            Compilation compilation, CancellationToken cancellationToken)
-        {
-            var groups = analysisResults.Select(r => (typeDeclaration: (TypeDeclarationSyntax)r.PropertyDeclaration.Parent!, r.SemanticModel))
-                                        .Distinct()
-                                        .GroupBy(n => n.typeDeclaration.SyntaxTree);
```

Comment:
walk of the entire type decl is unnecessary.  we register to hear about codeblocks directly inside the type symbol.  so we can just walk those trivially.